File: src/main/java/org/entities/Autonomous.java
package org.entities;

public abstract class Autonomous extends Entity{
    protected int state=0;//Controls various states, such as normal movement and melee movement
    public Autonomous(int subLevel,float spawnX,float spawnY){
        this.subLevel=subLevel;
        x=spawnX;
        y=spawnY;
    }
}

File: src/main/java/org/entities/Autonomous.kt
package org.entities

abstract class Autonomous(subLevel: Int, spawnX: Float, spawnY: Float) : Entity() {
    @JvmField
    var state = 0 //Controls various states, such as normal movement and melee movement

    init {
        this.subLevel = subLevel
        x = spawnX
        y = spawnY
    }
}